plugins/discovery/discovery_test.go

```diff
@@ -478,6 +478,71 @@ func TestReconfigure(t *testing.T) {
 
 }
 
+func TestReconfigureLocalOverride(t *testing.T) {
+
+	manager, err := plugins.New([]byte(`{
+		"labels": {"x": "y"},
+		"services": {
+			"localhost": {
+				"url": "http://localhost:9999"
+			}
+		},
+		"decision_logs": {"reporting": {"buffer_size_limit_bytes": 512}},
+		"discovery": {"name": "config"},
+		"caching": {"inter_query_builtin_cache": {"max_size_bytes": 10000000}},
+	}`), "test-id", inmem.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testPlugin := &reconfigureTestPlugin{counts: map[string]int{}}
+	testFactory := testFactory{p: testPlugin}
+
+	disco, err := New(manager, Factories(map[string]plugins.Factory{"test_plugin": testFactory}))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.Background()
+
+	initialBundle := makeDataBundle(1, `
+		{
+			"config": {
+				"labels": {"x": "label value changed"},
+				"default_decision": "bar/baz",
+				"default_authorization_decision": "baz/qux",
+				"plugins": {
+					"test_plugin": {"a": "b"}
+				},
+				"decision_logs": {"partition_name": "bar"}
```

Comment:
Would it be worth testing that in the case that discovery also sets the value, the local override wins?

Reply:
Yes, also we have to agree on order of the override, which is why I posted this as draft. The current version lets the local version win in case both discovery and local for example contain `decision_logs.runtime.*` configs. We can also reverse this. This still allows to generally configure operational attributes locally if they are not forced centrally.